DE-3223 Added get wikis api

jcellary opened the pull request:
Followup of https://github.com/Wikia/app/pull/14272

Added api for importing list of all wikis and tags for a wiki

Commit:
Added get wikis api

Commit:
Added max limit value

jcellary commented on includes/wikia/api/DWDimensionApiControllerSQL.class.php:
```diff
@@ -0,0 +1,58 @@
+<?php
+
+class DWDimensionApiControllerSQL {
```
@macbre Is this a reasonable way to include SQL or is there some other preferred way?

macbre commented on includes/wikia/api/DWDimensionApiControllerSQL.class.php:
```diff
@@ -0,0 +1,58 @@
+           city.city_dbname        AS dbname,
+           city.city_sitename      AS sitename,
+           REPLACE(REPLACE(city.city_url, \'http://\', \'\'), \'/\', \'\') AS url,
+           REPLACE(REPLACE(city.city_url, \'http://\', \'\'), \'/\', \'\') AS domain,
```
Can't we perform this logic in PHP?

macbre commented on includes/wikia/api/DWDimensionApiControllerSQL.class.php:
```diff
@@ -0,0 +1,58 @@
+                WHEN cat.cat_name = \'Gaming\'                    THEN \'Gaming\'
+                WHEN cat.cat_name = \'Wikia\'                     THEN \'Wikia\'
+                WHEN cat.cat_name = \'Lifestyle\'                 THEN \'Lifestyle\'
+                ELSE \'Lifestyle\' END        AS vertical_name,
```
`WikiFactory` class has a few, well cached methods for getting these data, such as category or vertical names.

Commit:
Improved vertical mapping

Commit:
Added dart tag api

Commit:
Fixed dart api

jcellary commented:
@macbre I replaced use cases with WikiFactory. Will still need to confirm with analytics if the default vertical mapping is ok. I also added a new api for dart and removed one that won't be used. Please take a look if it's good and if yes we could merge this part.

macbre commented on includes/wikia/api/DWDimensionApiController.class.php:
```diff
@@ -52,11 +52,16 @@ public function getWikiDartTags() {
+					'tag' => $r[ 1 ][ $i ],
+					'value' => $r[ 2 ][ $i ]
+				];
+			}
```
Don't we want to move WikiFactory to a service? This adds a dependency on a database. Let's use `WikiFactory` class helpers to get a variable value. It's pretty well cached.

jcellary replied:
WikiFactory in it's current form doesn't have a method to fetch values for all wikis, only a single one (at least from what I have seen). I was thinking about adding one, but wasn't sure if it's worth it. Do you think I should do that? Or is there some existing mechanism that I missed?

jcellary replied:
Ok moved to WikiFactory

Commit:
Moved variable fetching to WikiFactory

Commit:
Style fixes

Commit:
Style fixes

macbre commented on extensions/wikia/WikiFactory/WikiFactory.php:
```diff
@@ -3432,4 +3432,40 @@ static private function parseValue( $value, $type ) {
+	/**
+	 * Returns value of a given variable for all wikis
+	 *
+	 * @param $variableId String Id of variable to get
```
That's actually `int`, right?

jcellary replied:
true

macbre commented on extensions/wikia/WikiFactory/WikiFactory.php:
```diff
@@ -3432,4 +3432,40 @@ static private function parseValue( $value, $type ) {
+	 * Returns value of a given variable for all wikis
+	 *
+	 * @param $variableId String Id of variable to get
+	 * @param $limit String Limit of rows
```
The same here

jcellary replied:
Not sure, because we get this from query param, so I assume everything is a string (although obviously it contains a number). Should I do parsing in such case or just pass it along?

macbre commented on includes/wikia/api/DWDimensionApiController.class.php:
```diff
@@ -2,36 +2,101 @@
+		$limit = min($db->strencode( $this->getRequest()->getVal( 'wiki_limit', static::LIMIT ) ), static::LIMIT_MAX);
+		$afterWikiId = $db->strencode( $this->getRequest()->getVal( 'after_wiki_id', static::WIKIS_AFTER_WIKI_ID ) );
+
+		$variables = WikiFactory::getVariableForAllWikis( static::DART_TAG_VARIABLE_ID, $limit, $afterWikiId );
```
Let's use `WikiFactory::getVarIdByName` instead of hard-coding variable ID.

jcellary replied:
done

Commit:
Changed to using variable name

Commit:
Changed query params to int

jcellary commented:
Merging per offline conversation with Macbre
